api/src/main/java/edu/cornell/mannlib/vitro/webapp/searchengine/solr/SolrSearchEngine.java

```diff
@@ -199,13 +233,97 @@ public int documentCount() throws SearchEngineException {
 		return (int) response.getResults().getNumFound();
 	}
 
+	private CloseableHttpClient buildHttpClient() {
+		HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
+
+		// TODO: configure settings from properties
+		httpClientBuilder.setMaxConnPerRoute(100);
+		httpClientBuilder.setMaxConnTotal(100);
+		httpClientBuilder.setRetryHandler(new StandardHttpRequestRetryHandler(1, false));
+
+		return httpClientBuilder.build();
+	}
+
+	private SolrClient buildSolrClient(CloseableHttpClient httpClient) {
+		HttpSolrClient.Builder builder = new HttpSolrClient.Builder(solrServerUrl);
+
+		// TODO: configure settings from properties
+		builder.withSocketTimeout(10000); // socket read timeout
+		builder.withConnectionTimeout(10000);
+		builder.withHttpClient(httpClient);
+
+		return builder.build();
+	}
+
+	private ConcurrentUpdateSolrClient buildConcurrentUpdateSolrClient() {
+		ConcurrentUpdateSolrClient.Builder updateBuilder =
+				new ConcurrentUpdateSolrClient.Builder(solrServerUrl);
+
+		// TODO: configure settings from properties
+		updateBuilder.withConnectionTimeout(10000);
+		// no apparent 7.4.0 analogy to `setPollQueueTime(25)`
+
+		return updateBuilder.build();
+	}
+
+	private boolean coreExists() throws SolrServerException, IOException {
+		NamedList<String> params = new NamedList<>();
+		params.add("wt", "json");
+		params.add("action", "STATUS");
+		params.add("core", solrCore);
+		
+		GenericSolrRequest systemRequest = new GenericSolrRequest(METHOD.GET, "/admin/cores", params.toSolrParams());
+		NamedList<Object> systemResponse = queryEngine.request(systemRequest);
+
+		log.info(format("Solr core %s status; response = '%s'.", solrCore, systemResponse));
+
+		return Optional.ofNullable(systemResponse.get("status"))
+			.map(NamedList.class::cast)
+			.map(status -> status.get(solrCore))
+			.map(NamedList.class::cast)
+			.map(status -> status.get("name"))
+			.isPresent();
+	}
+
+	private void initializeCore(String contextPath) throws SolrServerException, IOException {
+		Path solrConfPath = Paths.get(contextPath + File.separator + "solr");
+		File solrConf = solrConfPath.toFile();
+
+		solrConf.setReadable(true, false);
```

Comment:
These are both returning `false`... not surprisingly.

Reply:
We may need to document a prerequisite of setting permissions on the Solr directories before start-up.

Reply:
Both users need to be able to write to the core directories: 
- solr user
- tomcat user

Reply:
I tried changing the owner and that resulted in AccessDeniedException. I am open to ideas with permission issues.

Reply:
At the point of requiring changing directory permissions, it is not much more to copy the conf directory and create the core.